compiler/types.nim

```diff
@@ -20,6 +20,7 @@ type
     preferName, preferDesc, preferExported, preferModuleInfo, preferGenericArg
 
 proc typeToString*(typ: PType; prefer: TPreferedDesc = preferName): string
+template `$`*(typ: PType): string = typeToString(typ)
```

Comment:
this causes problems with older Nim releases (because of bugs in them), better make it an .inline proc.